src/hooks/useCreateUser.tsx

```diff
@@ -0,0 +1,49 @@
+// src/hooks/useCreateUser.ts
+import { useState } from 'react';
+import { API_PATH } from '@/utils/constants';
+import { User } from '@/interfaces/User'; // Import the User interface
+
+// Define the input type for creating a user, excluding fields that are auto-generated like created_at and updated_at.
+type CreateUserInput = Omit<User, 'user_id' | 'created_at' | 'updated_at'> & {
+  password: string; // Include password separately since it’s used for creating users
+};
+
+const useCreateUser = () => {
+  const [userId, setUserId] = useState<string | null>(null); // Only store the user ID
```

Comment:
Rather than attempting to return the userId through a useState hook, I would actually through the API on successful post return the user's ID within the response body from the API itself. This saves overhead with the hook and lets the API takes care of what to return instead. The hook should only worry about sending a request and returning the response from the API.